src/main/java/de/dennisguse/opentracks/io/file/importer/KmzTrackImporter.java

```diff
@@ -42,58 +46,197 @@
 
     private static final String TAG = KmzTrackImporter.class.getSimpleName();
 
+    public static final List<String> KMZ_IMAGES_EXT = new ArrayList<>();
+    static {
+        KMZ_IMAGES_EXT.add("jpeg");
+        KMZ_IMAGES_EXT.add("jpg");
+        KMZ_IMAGES_EXT.add("png");
+    }
+
     private static final int BUFFER_SIZE = 4096;
 
     private final Context context;
     private final long importTrackId;
+    private Uri uriKmzFile;
 
     /**
      * Constructor.
      *
      * @param context       the context
      * @param importTrackId track id to import to. This should not be -1L so that images in the kmz file can be imported.
+     * @param uriFile       URI of the kmz file.
      */
-    KmzTrackImporter(Context context, long importTrackId) {
+    KmzTrackImporter(Context context, long importTrackId, Uri uriFile) {
         this.context = context;
         this.importTrackId = importTrackId;
+        this.uriKmzFile = uriFile;
     }
 
     @Override
     public long importFile(InputStream inputStream) {
-        long trackId = importTrackId;
+        long trackId;
+
+        if (!copyKmzImages()) {
+            cleanImport(context, importTrackId);
+            return -1L;
+        }
+
+        trackId = findAndParseKmlFile(inputStream);
+        if (trackId == -1L) {
+            cleanImport(context, importTrackId);
+            return -1L;
+        }
+
+        deleteOrphanImages(context, trackId);
+
+        return trackId;
+    }
+
+    /**
+     * Copies all images that are inside KMZ to OpenTracks external storage.
+     *
+     * @return false if there are errors or true otherwise.
+     */
+    private boolean copyKmzImages() {
+        try (InputStream inputStream = context.getContentResolver().openInputStream(uriKmzFile);
+            ZipInputStream zipInputStream = new ZipInputStream(inputStream)) {
+            ZipEntry zipEntry;
```

Comment:
Move into while?

Reply:
What do you mean to move into?  ZipEntry declaration?